src/AndersonAcceleration.f90

```diff
@@ -196,59 +143,217 @@ SUBROUTINE init_AndersonAccelerationType(solver,MPIEnv,Params)
+      solver%A(1,1)=0.0_SRK
+      solver%b(1)=0.0_SRK
+      DO j=1,solver%N
+        solver%A(1,1)=solver%A(1,1)+(solver%r(j,2)-solver%r(j,1))**2_SIK
+        solver%b(1)=solver%b(1)+solver%r(j,2)*(solver%r(j,2)-solver%r(j,1))
+      ENDDO
+      solver%alpha(1)=solver%b(1)/solver%A(1,1)
+    ELSEIF(depth_s > 1) THEN
+      !Construct coefficient matrix, right hand side, and solve for fit coefficients
+      DO i=1,depth_s
+        solver%b(i)=0.0_SRK
+        DO j=1,solver%N
+          solver%b(i)=solver%b(i)+solver%r(j,depth_s+1)*(solver%r(j,depth_s+1)-solver%r(j,i))
+        ENDDO
+        DO k=1,i
+          solver%A(k,i)=0.0_SRK
+          DO j=1,solver%N
+            solver%A(k,i)=solver%A(k,i)+(solver%r(j,depth_s+1)-solver%r(j,k))* &
+                                        (solver%r(j,depth_s+1)-solver%r(j,i))
+          ENDDO
+          IF(i /= k)solver%A(i,k)=solver%A(k,i)
+        ENDDO
+      ENDDO
+      solver%alpha(1:depth_s)=GECP(solver%A(1:depth_s,1:depth_s),solver%b(1:depth_s))
+    ENDIF
+    !Back out "0th" alpha
+    solver%alpha(depth_s+1)=0.0_SRK
+    DO i=1,depth_s
+      solver%alpha(depth_s+1)=solver%alpha(depth_s+1)+solver%alpha(i)
+    ENDDO
+    solver%alpha(depth_s+1)=1.0_SRK-solver%alpha(depth_s+1)
+
+    !Get accelerated solution
+    x_tmp(:)=0.0_SRK
+    DO i=1,depth_s+1
+      DO j=1,solver%N
+        x_tmp(j)=x_tmp(j)+solver%alpha(i)*(solver%x(j,i)+solver%beta*solver%r(j,i))
+      ENDDO
+    ENDDO
+
+    !Push back solution vectors and load in newest accelerated as next initial iterate
+    DO i=MIN(depth_s+1,solver%depth),1,-1
+      solver%x(:,i+1)=solver%x(:,i)
+    ENDDO
+    solver%x(:,1)=x_tmp(:)
   ENDIF
 
-  solver%iter=solver%iter+1
-#endif
-ENDSUBROUTINE step_AndersonAccelerationType
+ENDFUNCTION step_AndersonAccelerationType
 !
 !-------------------------------------------------------------------------------
-!> @brief Reset the Anderson acceleration solver
+!> @brief Set the initial iterate for the Anderson acceleration solver
 !> @param solver The anderson solver to act on
+!> @param x initial iterate
 !>
-!> This routine resets the counters for the anderson acceleration
-!>
-SUBROUTINE reset_AndersonAccelerationType(solver)
+SUBROUTINE set_AndersonAccelerationType(solver,x)
   CLASS(AndersonAccelerationType),INTENT(INOUT) :: solver
-#ifdef FUTILITY_HAVE_Trilinos_NOX
-  CALL Anderson_Reset(solver%id)
-  solver%iter=1
-#endif
-ENDSUBROUTINE reset_AndersonAccelerationType
+  REAL(SRK),INTENT(IN) :: x(:)
+
+  solver%x(:,1)=x(:)
+
+ENDSUBROUTINE set_AndersonAccelerationType
+!
+!-------------------------------------------------------------------------------
+!> @brief Function which solves the system Ax=b by Gaussian elimination with
+!>        full(complete) pivoting. NOTE that A is destroyed in the process.
+!> @param A coefficient matrix in Ax=b
+!> @param b right hand side of problem Ax=b
+!> @returns x the solution vector of the problem Ax=b
+!
+FUNCTION GECP(A,b) RESULT(x)
```

Comment:
"Gaussian Elimination Complete Pivoting" I'm guessing?  If so, 2 comments:
1. I _hate_ these super shortened names...
2. Don't we have this implemented in Futility using MatrixTypes and VectorTypes?  Really should use those insteaad.

Reply:
while I'm not convinced full pivoting is there, its unclear if its necessary.  And if it is needed and we don't have it, we should implement it in linear solvers.

Reply:
1. I hate super long names.
2. Matrix types / vector types don't have it specifically, but I doubt it's necessary to do full pivoting. Mostly I just wanted something I could drop in that's easy to use and get an answer instead of spending a bunch of time figuring out how to use all this vector and matrix types stuff.